Source/Diagnostics/WarpXOpenPMD.cpp

```diff
@@ -746,30 +754,39 @@ WarpXOpenPMDPlot::SetupPos(
 // this is originally copied from FieldIO.cpp
 //
 void
-WarpXOpenPMDPlot::WriteOpenPMDFields( //const std::string& filename,
+WarpXOpenPMDPlot::WriteOpenPMDFields ( //const std::string& filename,
                       const std::vector<std::string>& varnames,
                       const amrex::MultiFab& mf,
-                      const amrex::Geometry& geom,
+                      const amrex::Geometry& geom, // geometry of the mf/Fab
                       const int iteration,
-                      const double time ) const
+                      const double time, bool isBTD,
+                      const amrex::Geometry& full_BTD_snapshot ) const
 {
   //This is AMReX's tiny profiler. Possibly will apply it later
   WARPX_PROFILE("WarpXOpenPMDPlot::WriteOpenPMDFields()");
 
   AMREX_ALWAYS_ASSERT_WITH_MESSAGE(m_Series != nullptr, "openPMD series must be initialized");
 
+  amrex::Geometry full_geom = geom;
+  if( isBTD )
+      full_geom = full_BTD_snapshot;
+
+  bool first_BTD_write = false;
```

Comment:
Commenting this because the naming could be confusing for someone reading this as regular "full" diagnostics flush:
```suggestion
  // is this either a regular write (true) or the first write in a
  // backtransformed diagnostic (BTD):
  bool first_BTD_write = false;
```